tests/test_worker.py

```diff
@@ -700,21 +700,58 @@ def test_worker_sets_job_status(self):
         self.assertEqual(job.is_failed, True)
 
     def test_job_dependency(self):
-        """Enqueue dependent jobs only if their parents don't fail"""
-        q = Queue()
-        w = Worker([q])
-        parent_job = q.enqueue(say_hello, result_ttl=0)
+        """Enqueue dependent jobs only when appropriate"""
+        q = Queue(connection=self.testconn)
+        w = SimpleWorker([q], connection=q.connection)
+
+        # enqueue dependent job when parent successfully finishes
+        parent_job = q.enqueue(say_hello)
         job = q.enqueue_call(say_hello, depends_on=parent_job)
         w.work(burst=True)
-        job = Job.fetch(job.id)
+        job = Job.fetch(job.id, connection=self.testconn)
         self.assertEqual(job.get_status(), JobStatus.FINISHED)
 
+        # don't enqueue dependent job when parent fails
         parent_job = q.enqueue(div_by_zero)
         job = q.enqueue_call(say_hello, depends_on=parent_job)
         w.work(burst=True)
-        job = Job.fetch(job.id)
+        job = Job.fetch(job.id, connection=self.testconn)
+        self.assertNotEqual(job.get_status(), JobStatus.FINISHED)
+
+        q.empty()
+
+        # don't enqueue dependent job when Dependency.allow_failure=False (the default)
+        parent_job = q.enqueue(div_by_zero)
+        dependency = Dependency(jobs=parent_job)
+        job = q.enqueue_call(say_hello, depends_on=dependency)
+        w.work(burst=True)
+        job = Job.fetch(job.id, connection=self.testconn)
         self.assertNotEqual(job.get_status(), JobStatus.FINISHED)
 
+        # enqueue dependent job when Dependency.allow_failure=True
+        parent_job = q.enqueue(div_by_zero)
+        dependency = Dependency(jobs=parent_job, allow_failure=True)
+        job = q.enqueue_call(say_hello, depends_on=dependency)
+        w.work(burst=True)
+        job = Job.fetch(job.id, connection=self.testconn)
+        self.assertEqual(job.get_status(), JobStatus.FINISHED)
+
+        # only enqueue dependent job when all dependencies have finished/failed
+        first_parent_job = q.enqueue(div_by_zero)
+        second_parent_job = q.enqueue(long_running_job, args=(5,))
```

Comment:
Can we avoid using `long_running_job` for test cases where they're not needed? This will make tests run for longer.

Reply:
What should this be replaced with here then?